src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/steps.tsx

```diff
@@ -22,24 +25,88 @@ import { RawText, Text } from 'components/Text'
 import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
 import type { Asset } from 'lib/asset-service'
 import type { SwapperName } from 'lib/swapper/types'
+import { assertUnreachable } from 'lib/utils'
 
 import { SwapperIcon } from '../../TradeInput/components/SwapperIcon/SwapperIcon'
 import { TradeType } from '../types'
 
 const stepIcon = <StepIcon />
 const spinner = <Spinner />
 
+const getStatusIcon = (txStatus: TxStatus) => {
+  // TODO: proper light/dark mode colors here
+  switch (txStatus) {
+    case TxStatus.Confirmed:
+      return (
+        <Circle bg='green.500' size='100%'>
+          <CheckIcon />
+        </Circle>
+      )
+    case TxStatus.Failed:
+      return (
+        <Circle bg='red.500' size='100%'>
+          <CloseIcon p={1} />
+        </Circle>
+      )
+    // when the trade is submitting, treat unknown status as pending so the spinner spins
+    case TxStatus.Pending:
+    case TxStatus.Unknown:
+    default:
+      return (
+        <Circle bg='gray.750' size='100%'>
+          <Spinner />
+        </Circle>
+      )
+  }
+}
+
+const getChainShortName = (chainId: KnownChainIds) => {
+  switch (chainId) {
+    case KnownChainIds.AvalancheMainnet:
+      return 'AVA'
+    case KnownChainIds.OptimismMainnet:
+      return 'OP'
+    case KnownChainIds.EthereumMainnet:
+      return 'ETH'
+    case KnownChainIds.PolygonMainnet:
+      return 'POLY'
+    case KnownChainIds.GnosisMainnet:
+      return 'GNO'
+    case KnownChainIds.BnbSmartChainMainnet:
+      return 'BNB'
+    case KnownChainIds.ArbitrumMainnet:
+      return 'ARB'
+    case KnownChainIds.ArbitrumNovaMainnet:
+      return 'ARB-Nova'
+    case KnownChainIds.BitcoinMainnet:
+      return 'BTC'
+    case KnownChainIds.BitcoinCashMainnet:
+      return 'BCH'
+    case KnownChainIds.CosmosMainnet:
+      return 'COSM'
+    case KnownChainIds.ThorchainMainnet:
+      return 'THOR'
+    case KnownChainIds.DogecoinMainnet:
+      return 'DOGE'
+    case KnownChainIds.LitecoinMainnet:
+      return 'LTC'
```

Comment:
appreciate any input or suggestions to correct or improve these

Reply:
These all seem reasonable for now!